Applications/mnist/Tensorflow/Training_Keras.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python
+#
+# Copyright (C) 2020 Jijoong Moon <jijoong.moon@samsung.com>
+#
+# SPDX-License-Identifier: Apache-2.0-only
+#
+# @file	mnist_Keras.py
+# @date	13 July 2020
+# @brief	This is Simple Classification Example using Keras
+# @see		https://github.com/nnstreamer/nntrainer
+# @author	Jijoong Moon <jijoong.moon@samsung.com>
+# @bug		No known bugs except for NYI items
+#
+# mnist example
+# inputlayer -> conv2d (5x5) 6 filters -> pooling2d 2x2 (valid) -> conv2d (5x5) 12 filters ->
+#               pooling2d 2x2 (valid) -> flatten -> fully connected layer -> 10 class
+#
+
+import random
+import struct
+import os
+import tensorflow as tf
+import numpy as np
+from tensorflow.keras.layers import Input, Dense, Conv2D, Flatten, AveragePooling2D
+from tensorflow.keras import models, layers, optimizers
+from tensorflow.keras.callbacks import Callback
+
+TOTAL_TRAIN_DATA_SIZE=100
+TOTAL_LABEL_SIZE=10
+TOTAL_VAL_DATA_SIZE=100
+FEATURE_SIZE=784
+batch_size =32
+Learning = True
+Test = False
+num_epoch = 1500
+
+##
+# @brief Callback to get the data from model during trining
+# @param[in] Callback Keras Callback object
+class History_LAW(Callback):
+    def on_train_begin(self, logs={}):
+        self.epoch=[]
+        self.weights =[]
+        self.history ={}
+        self.weights.append(self.model.layers[0].get_weights())
+    
+    def on_epoch_end(self, epoch, logs={}):
+        self.weights.append(self.model.layers[0].get_weights())
+
+##
+# @brief input data generater with batch_size
+#        ( 32 x 28 x 28 x 1 :training & 32 x 10 : labels )
+# @param[in] x_data : total input data
+# @param[in] y_data : total label data
+# @param[in] batch_size : batch_size
+# @return (x_batch, y_batch)
+def datagen( x_data, y_data, batch_size):
+    size=len(x_data)
+    while True:
+        idx = np.random.permutation(size)
+        x_data=x_data[idx]
+        y_data=y_data[idx]
+
+        for i in range(size // batch_size):
+            x_batch = x_data[i*batch_size: (i+1)*batch_size]
+            x_batch=np.reshape(x_batch, (batch_size, 1, 28,28))
+            x_batch=np.transpose(x_batch, [0,2,3,1])
+            y_batch = y_data[i*batch_size: (i+1)*batch_size]
+            
+
+            yield x_batch, y_batch
+            
+##
+# @brief load input data from file
+# @return (InputVector, InputLabel, Validation Vector, ValidationLabel)
+def load_data():
+    tf.compat.v1.set_random_seed(777)
```

Comment:
this will go away once above comment is applied.

Reply:
done. Thanks